[high prio] enable VLLM_SPYRE_ENABLE_PREFILL_OPTIMIZATION by default

yannicks1 opened the pull request:
Our bench marking has shown that from both a user experience and a system throughput perspective this feature brings great benefits across all traces we evaluated on. 

this PR turns the feature on by default.

yannicks1 commented:
Changing the default value for this changes the processing in /test_spyre_cb_scheduler_steps.py. I can adapt those later when I have some slack time. If we wanna get this in asap, feel free to force merge @tdoublep 
cc @joerunde .

sducouedic commented on tests/e2e/test_spyre_cb_scheduler_steps.py:
```diff
@@ -686,8 +686,8 @@ def test_prompt_too_long_for_current_tkv(model: ModelInfo, backend: str,
 
-    if prefill_optimization:
-        monkeypatch.setenv('VLLM_SPYRE_ENABLE_PREFILL_OPTIMIZATION', '1')
+    if not prefill_optimization:
+        monkeypatch.setenv('VLLM_SPYRE_ENABLE_PREFILL_OPTIMIZATION', '0')
```
Maybe it would be better to not assume any default value and set it anyway. It makes things more explicit.

```suggestion
    monkeypatch.setenv('VLLM_SPYRE_ENABLE_PREFILL_OPTIMIZATION', '1' if prefill_optimization else '0')
```

sducouedic commented on tests/e2e/test_spyre_cb_scheduler_steps.py:
```diff
@@ -913,8 +913,6 @@ def test_prefill_optimization_tkv_too_big(model: ModelInfo, backend: str,
             * 1: len = 70, max tokens = 50, step joining = 0
     """
 
-    monkeypatch.setenv('VLLM_SPYRE_ENABLE_PREFILL_OPTIMIZATION', '1')
-
```
Same, why not keeping it to make it explicit

sducouedic commented on tests/e2e/test_spyre_cb_scheduler_steps.py:
```diff
@@ -1084,8 +1082,6 @@ def test_prefill_optimization_use_more_than_available_blocks(
         * available_blocks: 4
     """
 
-    monkeypatch.setenv('VLLM_SPYRE_ENABLE_PREFILL_OPTIMIZATION', '1')
```
same

sducouedic commented on tests/e2e/test_spyre_cb_scheduler_steps.py:
```diff
@@ -1942,14 +1940,15 @@ def test_requests_exceed_batch_tkv_limit(model: ModelInfo, backend: str,
-        model: ModelInfo, backend: str, monkeypatch: pytest.MonkeyPatch,
-        set_random_seed, max_num_seqs: int, max_model_len: int,
-        available_blocks: int):
+def test_requests_use_full_batch_tkv_limit(model: ModelInfo, backend: str,
```
Some other tests above are prefixed with "test_prefill_optimization_"

sducouedic reviewed (commented):
In my opinion it is better not to assume any default value for the prefill optimization. All the tests that have different behaviour with and without would be named "{test_name}_prefill_opt", "{test_name}_no_prefill_opt" and environment variable set accordingly

yannicks1 commented:
Hi @sducouedic I addressed your feedback: 
All the tests that have different behaviour with and without the optimization are now named "{test_name}_prefill_opt" and "{test_name}_no_prefill_opt" and the environment variable set explicitly. The rest of the test are untouched. 

Note that tests starting with "test_prefill_optimization_" do explicitly test the prefill optimization (I also explicitly set the env there).
